src/dir.rs

```diff
@@ -200,20 +200,33 @@ impl<'a, IO: ReadWriteSeek, TP: TimeProvider, OCC: OemCpConverter> Dir<'a, IO, T
     ///
     /// `path` is a '/' separated directory path relative to self directory.
     ///
+    /// An empty path returns the current directory.
+    /// A trailing slash is ignored.
+    /// (But the path '/' will not be found.)
+    ///
     /// # Errors
     ///
     /// Errors that can be returned:
     ///
     /// * `Error::NotFound` will be returned if `path` does not point to any existing directory entry.
     /// * `Error::InvalidInput` will be returned if `path` points to a file that is not a directory.
     /// * `Error::Io` will be returned if the underlying storage object returned an I/O error.
-    pub fn open_dir(&self, path: &str) -> Result<Self, Error<IO::Error>> {
+    pub fn open_dir(&self, mut path: &str) -> Result<Self, Error<IO::Error>> {
         trace!("Dir::open_dir {}", path);
-        let (name, rest_opt) = split_path(path);
-        let e = self.find_entry(name, Some(true), None)?;
-        match rest_opt {
-            Some(rest) => e.to_dir().open_dir(rest),
-            None => Ok(e.to_dir()),
+        let mut dir = self.clone();
+        loop {
+            if path.is_empty() {
+                return Ok(dir);
+            }
+            let (name, rest_opt) = split_path(path);
+            let e = dir.find_entry(name, Some(true), None)?.to_dir();
```

Comment:
Name `e` no longer make sense here. Please remove it and update `dir` instead

Reply:
updated